lib/guardian/plug/backdoor.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Guardian.Plug.Backdoor do
+  @moduledoc """
+  This plug allows you to bypass authentication in acceptance tests
+  by passing the token needed to load the current resource directly to your
+  serializer, via a query string parameter.
+
+  ## Installation
+
+  Add the following to your Phoenix router before any other Guardian plugs.
+
+  ```
+  if Mix.env == :test do
+    plug Guardian.Plug.Backdoor
+  end
+  ```
+
+  ***This plug is designed for acceptance testing scenarios and should
+  never be added in a production environment.***
+
+  ## Usage
+
+  Now that `Guardian.Plug.Backdoor` is installed, it's time to log in.
+  If you're using [Hound][hound], you can write the following code.
+
+  ```
+  navigate_to "/?as=User:5"
+  ```
+
+  If you aren't using [Hound][hound], a simple `GET /?as=User:5` request will
+  work.
+
+  When the `Guardian.Plug.Backdoor` plug runs, it passes along the value
+  of the `as` parameter directly to your application's Guardian serializer.
+
+  ## Options
+
+  The following options can be set when instantiating the plug.
+
+  * `serializer` - The serializer to be used to load the current resource.
+    Defaults to the serializer configured in your app's Guardian config.
+  * `token_field` - Query string field used to load the current resource.
+    Defaults to `as`.
+  * `type` - Type of token, passed directly to Guardian.Plug.sign_in/4.
+  * `new_claims` - New claims to be encoded in the JWT, passed directly to
+    Guardian.Plug.sign_in/4.
+
+  [hound]: https://github.com/HashNuke/hound
+  """
+  import Plug.Conn
+
+  @doc false
+  def init(opts \\ []) do
```

Comment:
@hassox I don't even see a need for this in `:dev`, do you?  I can see someone mistakenly deploying their code without setting MIX_ENV=prod and this would leak.  Thoughts?